api.go

```diff
@@ -659,9 +659,7 @@ func (c Client) newRequest(method string, metadata requestMetadata) (req *http.R
 		// Set sha256 sum for signature calculation only with signature version '4'.
 		shaHeader := unsignedPayload
 		if !c.secure {
-			if metadata.contentSHA256Bytes == nil {
-				shaHeader = hex.EncodeToString(sum256([]byte{}))
-			} else {
+			if len(metadata.contentSHA256Bytes) > 0 {
```

Comment:
@krishnasrinivas It is _not_ wrong to compute sha256 sum of `[]byte{}` and set `x-amz-content-sha256` header with it. See `Important section` here  http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html. `UNSIGNED-PAYLOAD` is an alternate. So, the server should work with requests signed using either of the approaches.